cmd/ctl/pkg/install/helm/settings.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"context"
+
+	"github.com/cert-manager/cert-manager/cmd/ctl/pkg/factory"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/pkg/cli"
+)
+
+const defaultCertManagerNamespace = "cert-manager"
+
+type NormalisedEnvSettings struct {
+	EnvSettings *cli.EnvSettings
+	Factory     *factory.Factory
+}
+
+func NewNormalisedEnvSettings() *NormalisedEnvSettings {
+	return &NormalisedEnvSettings{
+		EnvSettings: cli.New(),
+	}
+}
+
+func (n *NormalisedEnvSettings) Namespace() string {
+	return n.Factory.Namespace
+}
+
+func (n *NormalisedEnvSettings) Setup(ctx context.Context, cmd *cobra.Command) {
+	n.Factory = factory.New(ctx, cmd)
+
+	// We add flags explicitly as the EnvSettings.AddFlags method gives them non normalised names.
+	// Since we will be generating the RestConfig from the Factory, not the EnvSettings, we can
+	// omit Kubernetes flags.
+	fs := cmd.Flags()
+	fs.BoolVar(&n.EnvSettings.Debug, "debug", n.EnvSettings.Debug, "enable verbose output")
+	fs.StringVar(&n.EnvSettings.RegistryConfig, "registry-config", n.EnvSettings.RegistryConfig, "path to the registry config file")
+	fs.StringVar(&n.EnvSettings.RepositoryConfig, "repository-config", n.EnvSettings.RepositoryConfig, "path to the file containing repository names and URLs")
+	fs.StringVar(&n.EnvSettings.RepositoryCache, "repository-cache", n.EnvSettings.RepositoryCache, "path to the file containing cached repository indexes")
```

Comment:
@ThatsMrTalbot Could the [SetNormalizeFunc](https://pkg.go.dev/github.com/spf13/pflag#FlagSet.SetNormalizeFunc) function fix this issue?

Reply:
I agree with that change, I was just wondering if we could use SetNormalizeFunc + some flag filtering to set these 4 flags in this file.

Reply:
https://github.com/cert-manager/cert-manager/blob/a10d81be5dd5d8ba1d3c973f54116f6d1bdcf042/pkg/logs/logs.go#L88-L96

Reply:
The VisitAll function in the example filters what flags to add, not what flags to skip.